BrushSnapshot::takeSnapshot: clear the Texture pointer in the snapshotted face

ericwa opened the pull request:
This is to avoid dangling pointers to Texture objects.
Also added a BrushTest.snapshotTextureTest to ensure this is being done.
Fixes https://github.com/kduske/TrenchBroom/issues/1520

kduske commented:
Doesn't the texture need to be set again after the snapshot was restored? And also, isn't the usage count increased when the face is cloned, so that now it is too high?

ericwa commented:
I think the texture usage counts are OK; cloning the face increases it, but the `faceClone->setTexture(nullptr);` I added decreases the usage count, so there is no net change. I added some assertions to the test for this.

>Doesn't the texture need to be set again after the snapshot was restored?

Looks like this is handled by `MapDocumentCommandFacade::restoreSnapshot` calling `MapDocument::setTextures`.. does that seem right?
It seems to work.

kduske commented:
Excellent. Maybe I'll add batest that ensures that the correct texture is set upon restore.
